### PR TITLE
Add webpack 5 support

### PR DESCRIPTION
**Pull request by bloodlit**

### CODE REVIEW

**Comment by devsuitup** [comment]
Thank you so much dude !

**Comment by icleolion** [comment]
Seeing as this loader doesn't claim to support webpack 3 I would take this PR over https://github.com/vuetifyjs/vuetify-loader/pull/158

**Comment by bloodlit** [comment]
> Seeing as this loader doesn't claim to support webpack 3 I would take this PR over #158

// webpack4 and lower
VuetifyLoaderPlugin = require('./plugin-webpack4')

If earlier Webpack 3 was provided, then it will work.

**Comment by hmnd** [comment]
> > Seeing as this loader doesn't claim to support webpack 3 I would take this PR over #158
> 
> // webpack4 and lower
> VuetifyLoaderPlugin = require('./plugin-webpack4')
> 
> If earlier Webpack 3 was provided, then it will work.

Your PR is basically a duplicate of #158 though. Why open a new one?

File: lib/plugin-webpack5.js [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        resourceQuery: '?',
+        use: rule.use
+      },
+      {
+        use: [
+          {
+            loader: require.resolve('./loader'),
+            options: {
+              match: this.options.match || [],
+              attrsMatch: this.options.attrsMatch || []
```
Comment (icleolion) [inline lib/plugin-webpack5.js:189]:
```suggestion
              attrsMatch: this.options.attrsMatch || [],
              registerStylesSSR: this.options.registerStylesSSR || false
```

Is `registerStylesSSR` deliberately not included in the webpack5 version or was this an oversight?

**Comment by icleolion** [comment]
@hmnd There are a handful of differences so if @bloodlit feels his attempt is better than #158 then why not raise a new PR.

@bloodlit my point about webpack 3 support is based on [vue-loader/commit/745d054](https://github.com/vuejs/vue-loader/commit/745d0542ef8ae6f6f740dfe2aeea1ba208ea293a#diff-11af61311ab29f43ed180a25e84cbae6a92c72a244b8c4f717d573ec930ff5ac).  They had to beef up that `if` check to support webpack 3 for some reason.  But vue-loader lists webpack 3 as one of it's peerDependencies, whereas vuetify-loader does not, so that extra check included in #158 is a waste of bytes.

Speaking of peerDependencies , neither this PR nor #158 update the peerDependencies to include support for webpack 5.

Also, there has been a number of developments to webpack 5 that vue-loader has had to keep up with.  I'm assuming vue-loader's implementation inspired this PR just like it did with #158, so do any of the below commits inspire further development on this PR to better support webpack 5?

1. [feat: support Rule.rules (nested rules) (#1618)](https://github.com/vuejs/vue-loader/commit/59433190d20bd864d9020168100904791b58cb0d#diff-4585872bb4a620b080a3ea3a07470fd5426a87b7f677b4490a11200e58a54a7d)
2. [fix: ensure unique `ident` when cloning rules, fix webpack 5 compatibility (#1653)](https://github.com/vuejs/vue-loader/commit/84c34a40eeeccb2ca39e390a03bdce13530e3c9e#diff-ae55ab2a1d46d7d2dc4e93047b3465cf1b6dea9cd63303a14dff7f978c02dfa4)
3. [fix: skip matching rule with 'enforce'](https://github.com/vuejs/vue-loader/commit/e7b2b1181ec2e67db87ba0895c9fb6c5608d5442#diff-ae55ab2a1d46d7d2dc4e93047b3465cf1b6dea9cd63303a14dff7f978c02dfa4)
4. [fix: ensure new webpack5 generator property in rules passes ruleset checks (#1753)](https://github.com/vuejs/vue-loader/commit/f79bb087f9211c3ea19d63c1dcfd758d6dbad095#diff-ae55ab2a1d46d7d2dc4e93047b3465cf1b6dea9cd63303a14dff7f978c02dfa4)
5. [fix: add rule plugins to keep up with webpack 5 new features](https://github.com/vuejs/vue-loader/commit/7ca8e2c58926144042095a67eea9326e4c206b13#diff-ae55ab2a1d46d7d2dc4e93047b3465cf1b6dea9cd63303a14dff7f978c02dfa4)

**Comment by petrprikryl** [comment]
Hi I am trying this PR on my project and have problem with styles order in **development**. My custom styles are being overriden by Vuetify styles:

image

`.v-select ` styles should be ignored as it should be loaded before my custom `.catalogue__sort`. Problem seems to be with **inline style blocks ordering** in dev mode. Very strange is that this is not happening after bundling and serving plain css without dev server. 

Next I tried to import whole `vuetify/dist/vuetify.min.css` and disable `vuetify-loader` and this problem wasn't happening. So I am suspect that the problem is here. Because I wasn't facing it with Webpack 4 and offical version 1.6.0 of `vuetify-loader`.

Can anybody push me the right way where the problem is?

File: lib/plugin-webpack5.js (hunk 1 above)
Comment (bloodlit) [reply lib/plugin-webpack5.js:189]: oversight

**Comment by martinoppitz** [comment]
I checked this PR - It build, but there a lot of errors in SASS.

-------

the same story here:
- https://github.com/vuetifyjs/vuetify-loader/issues/136 and here
- https://github.com/aurelia/aurelia/issues/982 and here
- https://github.com/sveltejs/svelte-loader/pull/136 and so on

Many thanks to Angular for influencing Webpack and the unconsolidated upgrade to v5.

**Comment by DRoet** [comment]
Checked this PR now that `@vue/cli-service": "5.0.0-alpha.0` is out, but it indeed throws a lot of SASS errors

**Comment by KaelWD** [comment]
Yeah I have my own implementation in progress that works with `progressiveImages` and has less code duplication. Thanks for the effort but it doesn't really cut it.

**Pull request closed without merging**